### PR TITLE
dev/core#1782 Set activity_date_time and created_date to default to current_timestamp

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Add defaults for `activity_date_time` and `created_date` if not set.

Before
----------------------------------------
No default.

After
----------------------------------------
Default

Technical Details
----------------------------------------

Comments
----------------------------------------
Found via https://github.com/civicrm/civicrm-core/pull/17274

@seamuslee001 I think we might need an upgrader step too so the database gets the defaults defined?

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I think the tests will do a good job of finding any issues here. Once the upgrade change is in too then then I'm happy when jenkins is

File: xml/schema/Activity/Activity.xml [hunk 1]
```diff
@@ -121,6 +121,9 @@
   <field>
     <name>activity_date_time</name>
     <import>true</import>
+    <required>false</required>
+    <export>true</export>
+    <default>DEFAULT CURRENT_TIMESTAMP</default>
```
Comment (seamuslee001) [inline xml/schema/Activity/Activity.xml:126]: just needs to be CURRENT_TIMESTAMP() I think?

**Comment by seamuslee001** [comment]
The failure was

```
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1064 (42000) at line 4202: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DEFAULT CURRENT_TIMESTAMP COMMENT 'Date and time this activity is scheduled to o' at line 8
``` 
Also note this

"Previously, at most one TIMESTAMP column per table could be automatically initialized or updated to the current date and time. This restriction has been lifted. Any TIMESTAMP column definition can have any combination of DEFAULT CURRENT_TIMESTAMP and ON UPDATE CURRENT_TIMESTAMP clauses. In addition, these clauses now can be used with DATETIME column definitions. For more information, see Automatic Initialization and Updating for TIMESTAMP and DATETIME." 

https://dev.mysql.com/doc/relnotes/mysql/5.6/en/news-5-6-5.html So this would break anyone who was trying to install on MySQL 5.5 (no of course deprecated but we still permit installation on it)

**Comment by eileenmcnaughton** [comment]
@colemanw any thoughts?

File: xml/schema/Activity/Activity.xml (hunk 1 above)
Comment (colemanw) [inline xml/schema/Activity/Activity.xml:126]:
```suggestion
    <default>CURRENT_TIMESTAMP</default>
```

File: xml/schema/Activity/Activity.xml [hunk 2]
```diff
@@ -466,7 +469,7 @@
     <comment>When was the activity was created.</comment>
     <required>false</required>
     <export>true</export>
-    <default>NULL</default>
+    <default>DEFAULT CURRENT_TIMESTAMP</default>
```
Comment (colemanw) [inline xml/schema/Activity/Activity.xml:472]:
```suggestion
    <default>CURRENT_TIMESTAMP</default>
```

**Comment by colemanw** [comment]
It sounds like we'll have to drop support for MySql 5.5 before releasing this.
That's fine with me but the question is how much lead time or warning should we give?
Looks like only [last month](https://github.com/civicrm/civicrm-core/pull/17261) we bumped the minimum from 5.1 to 5.5!
According to http://stats.civicrm.org/?tab=technology most sites are already up to or beyond 5.6, with about a thousand sites still running 5.5 (those same sites may well be running an older version of Civi as well).

**Comment by colemanw** [comment]
@mattwire the DAO still needs regen.
We'll also need to put in some upgrade sql but need to pick a target version first.
If we merge this in June then it goes into the RC in July and released in August; we'd just need to put the word out btw now and then that we are dropping 5.5 support.
We should also bump up `CRM_Upgrade_Incremental_General::MIN_INSTALL_MYSQL_VER` as part of this PR.

**Comment by colemanw** [comment]
Ref https://lab.civicrm.org/dev/core/-/issues/1681

**Comment by eileenmcnaughton** [comment]
We discussed this & plan is remove mysql 5.5 support in 5.28 - so with some test changes &  changes to support that we will be able to merge this

**Comment by mattwire** [comment]
@colemanw @eileenmcnaughton I've added the upgrader and the min SQL version. And regenerated DAO. Are we good to merge?

**Comment by colemanw** [comment]
Looks great!

**Comment by eileenmcnaughton** [comment]
I removed the ready for review as this is stale. I'm not a fan of the label since in theory they should all be if they are in the review queue

**Comment by mattwire** [comment]
@colemanw This is now rebased following DAO updates.

**Comment by eileenmcnaughton** [comment]
If this doesn't pass it might be that we need to adjust CI - @seamuslee001 emailed the dev list to highlight that we will discontinue 5.5 mysql support

**Comment by eileenmcnaughton** [comment]
nice!

**Comment by colemanw** [comment]
Nice work everyone. And thanks for the followup #17496 @seamuslee001

**Pull request merged**